src/app/data-planes/views/DataPlaneDetailView.vue

```diff
@@ -1,21 +1,313 @@
+          </KCard>
+        </div>
+      </div>
     </AppView>
   </RouteView>
 </template>
 
 <script lang="ts" setup>
-import DataPlaneDetails from '../components/DataPlaneDetails.vue'
+import { KUI_ICON_SIZE_30 } from '@kong/design-tokens'
+import { KAlert, KCard, KIcon, KTooltip } from '@kong/kongponents'
+import { computed } from 'vue'
+
+import { useCan } from '@/app/application'
+import DefinitionCard from '@/app/common/DefinitionCard.vue'
+import StatusBadge from '@/app/common/StatusBadge.vue'
+import SubscriptionList from '@/app/common/subscriptions/SubscriptionList.vue'
+import TagList from '@/app/common/TagList.vue'
+import {
+  parseMTLSData,
+} from '@/app/data-planes/data'
 import type { DataPlaneOverview } from '@/types/index.d'
+import { useI18n } from '@/utilities'
+import {
+  compatibilityKind,
+  COMPATIBLE,
+  dpTags,
+  getStatusAndReason,
+  getVersions,
+  INCOMPATIBLE_WRONG_FORMAT,
+} from '@/utilities/dataplane'
+
+const { formatIsoDate } = useI18n()
+const can = useCan()
 
 const props = defineProps<{
   data: DataPlaneOverview
 }>()
+
+const statusWithReason = computed(() => getStatusAndReason(props.data.dataplane, props.data.dataplaneInsight))
+const dataPlaneTags = computed(() => dpTags(props.data.dataplane))
+const dataPlaneVersions = computed(() => getVersions(props.data.dataplaneInsight))
+const mtlsData = computed(() => parseMTLSData(props.data))
+
+const warnings = computed(() => {
+  const subscriptions = props.data.dataplaneInsight?.subscriptions ?? []
+  if (subscriptions.length === 0) {
+    return []
+  }
+
+  const lastSubscription = subscriptions[subscriptions.length - 1]
+  if (!('version' in lastSubscription) || !lastSubscription.version) {
+    return []
+  }
+
+  const version = lastSubscription.version
+
+  const warnings: { kind: string, payload?: any }[] = []
+  if (version.kumaDp && version.envoy) {
+    const compatibility = compatibilityKind(version)
+
+    if (compatibility.kind !== COMPATIBLE && compatibility.kind !== INCOMPATIBLE_WRONG_FORMAT) {
+      warnings.push(compatibility)
+    }
+  }
+  if (
+    mtlsData.value &&
+    (Date.now() > new Date(mtlsData.value?.certificateExpirationTime).getTime())
+  ) {
+    warnings.push({
+      kind: 'CA_EXPIRED',
+      payload: {},
+    })
+  }
+
+  if (can('use zones')) {
+    const tags = dpTags(props.data.dataplane)
+    const zoneTag = tags.find(tag => tag.label === 'kuma.io/zone')
+
+    if (zoneTag && typeof version.kumaDp.kumaCpCompatible === 'boolean' && !version.kumaDp.kumaCpCompatible) {
+      warnings.push({
+        kind: 'INCOMPATIBLE_ZONE_CP_AND_KUMA_DP_VERSIONS',
+        payload: {
+          kumaDp: version.kumaDp.version,
+        },
+      })
+    }
+  }
+
+  return warnings
+})
+
 </script>
+<style lang="scss" scoped>
+.status-with-reason {
+  display: flex;
+  align-items: center;
+  gap: $kui-space-50;
+}
+</style>
+
+<style lang="scss">
+.reason-tooltip .kong-icon {
+  display: flex;
+  align-items: center;
+}
+</style>
```

Comment:
All of this green is because I removed the `DataPlaneDetail.vue` component entirely and copied its source directly into `DataPlaneDetailView.vue`